google/cloud/bigquery/job/query.py

```diff
@@ -1457,11 +1461,14 @@ def result(  # type: ignore  # (incompatible with supertype)
                 is ``DONE``, retrying is aborted early even if the
                 results are not available, as this will not change
                 anymore.
-            timeout (Optional[float]):
+            timeout (Optinal[Union[float, \
+                google.api_core.future.polling.PollingFuture._DEFAULT_VALUE, \
+            ]]):
                 The number of seconds to wait for the underlying HTTP transport
-                before using ``retry``.
-                If multiple requests are made under the hood, ``timeout``
-                applies to each individual request.
+                before using ``retry``. If ``None``, wait indefinitely
+                unless a retriable error is returned. If unset, only the
+                underlying Client.get_job() API call has timeout, but we still
```

Comment:
I'm not sure if we need to be this specific, as we do make other API calls and may want to set a default timeout there too (and do sometimes already in the case of jobs.getQueryResults)

```suggestion
                underlying API calls have their default timeouts, but we still
```